### PR TITLE
Consistently declare Protobuf enums

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Enums were not consistently declared. Some were at the global scope,
some others wire nested in messages. This moves all enums to the global
scope and uses most of the official style guide rules here:
https://developers.google.com/protocol-buffers/docs/style#enums

Unfortunately this results in long and stuttering names of enum values in
the code but this change is necessary to future-proof the proto and ensure
no enum names conflict when we add new enums and values (all enum value
names are in one namespace).

I don't like the end result but not doing it appears to be worse.
We have seen this first-hand in OTLP proto where enum names are
inconsistent and are now impossible to fix due to backward compatibility
concerns.

Note: I ignored the rule to CAPITALIZE names and use underscores
because if I do that the names in the code become unreadable walls
of text.

If anyone has any ideas how to fix the enum names issue in a nicer
way I am happy to discard this PR.

One alternate is to remove the prefixes from enum value names and
just require that all enum value names introduced in the future are
also unique, but I think it can result in weird names in the future.

### CODE REVIEW

File: internal/proto/opamp.proto [hunk 1]
```diff
@@ -450,21 +440,38 @@ message ServerErrorResponse {
...
+    // The AgentToServer message was malformed. The Agent SHOULD NOT retry
+    // the message.
+    ServerErrorResponse_BadRequest = 1;
+
+    // The Server is overloaded and unable to process the request. The Agent
+    // should retry the message later. retry_info field may be optionally
+    // set with additional information about retrying.
+    ServerErrorResponse_Unavailable = 2;
+}
+
```
Comment (andykellr) [inline internal/proto/opamp.proto:458]: Should these include Type, e.g. `ServerErrorResponseType_Unknown`, to match the name of the enum?

Comment (tigrannajaryan) [reply internal/proto/opamp.proto:458]: Yes, let me fix.

Comment (tigrannajaryan) [reply internal/proto/opamp.proto:458]: Fixed.

File: internal/proto/opamp.proto [hunk 2]
```diff
@@ -660,32 +668,33 @@ message PackageStatus {
...
-    }
-    Status status = 6;
+    PackageStatusEnum status = 6;
 
     // Error message if the status is erroneous.
     string error_message = 7;
 }
 
+// The status of this package.
+enum PackageStatusEnum {
```
Comment (andykellr) [inline internal/proto/opamp.proto:678]: nit: Should we call this PackageStatuses to match other similar enum names? This is the only one with an Enum suffix.

Comment (tigrannajaryan) [reply internal/proto/opamp.proto:678]: Unfortunately I can't because we already have a message named `PackageStatuses`. This was a forced deviation.

**Review by andykellr — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Rebased and fixed merge conflicts. @andykellr if you have time please take another look (just want to make sure I didn't miss anything), otherwise I will merge soon.

**Pull request merged**